src/couch_replicator_manager.erl

```diff
@@ -213,11 +216,13 @@ handle_cast(Msg, State) ->
     twig:log(error, "Replication manager received unexpected cast ~p", [Msg]),
     {stop, {error, {unexpected_cast, Msg}}, State}.
 
-handle_info({nodeup, _Node}, State) ->
-    {noreply, rescan(State)};
+handle_info({nodeup, Node}, State) ->
+    Live = lists:usort([Node|State#state.live]),
```

Comment:
where I think problems occur is that the ordering between nodes is random, and nodes() changes immediately. so two nodes scanning for the 'same' nodeup/down event might make different choices for owner(), resulting in all nodes evaluating to false but thinking they did the right thing. for that reason, keeping Live the same for the full scan was an attempt to eliminate that.